backend/handlers/handlers.go

```diff
@@ -339,6 +343,25 @@ func (handlers *Handlers) postAddAccountHandler(r *http.Request) (interface{}, e
 		return nil, err
 	}
 	keypath := signing.NewEmptyAbsoluteKeypath()
+
+	if jsonAddress != "" {
+		if !common.IsHexAddress(jsonAddress) {
+			return nil, errp.WithStack(errors.ErrInvalidAddress)
+		}
+
+		if jsonCoinCode == "teth" || jsonCoinCode == "eth" {
+			configuration := signing.NewAddressConfiguration(scriptType, keypath, jsonAddress)
+			getSigningConfiguration := func() (*signing.Configuration, error) {
```

Comment:
from here to line 358 is all duplicated code from, same lines appear below.

The only thing different is the configuration itself. So it makes sense to declare  `var configuration *signing.Configuration` above, and simply assign it here and below.